lib/Basics/Thread.cpp

```diff
@@ -201,7 +201,7 @@ void Thread::shutdown() {
   LOG_TOPIC("93614", TRACE, Logger::THREADS) << "shutdown(" << _name << ")";
 
   beginShutdown();
-  if (_threadStructInitialized) {
+  if (_threadStructInitialized && isRunning()) {
```

Comment:
This is problematic! - `isRunning` returns false if `_state == STOPPED`, but that does not necessarily mean that the thread has already terminated and was joined or detached.

Reply:
BTW: I just realized that on Windows we leak the thread handles. `TRI_JoinThreadWithTimeout` should call `CloseHandle` after waiting for the thread.